src/test/article.spec.js

```diff
@@ -304,3 +309,356 @@ describe('GET api/v1/articles', () => {
+    expect(response.status).to.equal(200);
+    expect(response.body.data.title).to.equal(article.title);
+    expect(response.body.data.description).to.equal(article.description);
+    expect(response.body.data.body).to.equal(article.body);
+    expect(response.body.data.image).to.equal(article.image);
+    expect(response.body.data.slug).to.equal(article.slug);
+    expect(response.body.data.slug).to.be.a('string');
+    expect(response.body.data.authorId).to.equal(article.authorId);
+    expect(response.body.data.publishedDate).to.equal(null);
+  });
+
+  it('should throw a 404 error when a specific article is not found', async () => {
+    const nonExistentArticleId = 5;
+    const response = await server().get(
+      `${ARTICLES_API}/${nonExistentArticleId}`
+    );
+    expect(response.status).to.equal(404);
+    expect(response.body.message).to.equal(
+      'The requested article was not found'
+    );
+  });
+
+  it('should return a 400 if articleId is invalid or not a non-zero positive an integer', async () => {
+    const invalidArticleId = '--gr444eat';
+    const response = await server().get(`${ARTICLES_API}/${invalidArticleId}`);
+    expect(response.status).to.equal(400);
+    expect(response.body.message).to.equal(
+      'Invalid Article ID. Article ID must be a positive integer'
+    );
+  });
+});
+
+describe('DELETE /api/v1/articles/:articleId', () => {
+  beforeEach(async () => {
+    await db.Bookmark.destroy({ cascade: true, truncate: true });
+    await db.User.destroy({ cascade: true, truncate: true });
+    await db.Article.destroy({ cascade: true, truncate: true });
+  });
+
+  it('should throw a 401 status code when creating an article with invalid token', async () => {
+    const invalidToken = 'sdsf89u023434n23knslfdasa.xzdfsdf';
+    const token = invalidToken;
+
+    const response = await server()
+      .delete(`${ARTICLES_API}/:articleId`)
+      .set('x-access-token', token)
+      .send(articleSample);
+    expect(response.status).to.equal(401);
+    expect(response.body.message).to.equal('Token is not valid');
+  });
+
+  it('should throw a 400 Your request cannot be processed right now, Please try again later when token is not provided', async () => {
+    const response = await server()
+      .delete(`${ARTICLES_API}/:articleId`)
+      .send(articleSample);
+    expect(response.status).to.equal(400);
+    expect(response.body.message).to.equal('Invalid access token');
+  });
+
+  it('should successfully delete an article', async () => {
+    const newUser = await createUser();
+    const token = helper.jwtSigner(newUser);
+
+    const validArticle = await BaseRepository.create(db.Article, articleSample);
+
+    const response = await chai
+      .request(app)
+      .delete(`${ARTICLES_API}/${validArticle.id}`)
+      .set('x-access-token', token);
+    expect(response.status).to.equal(200);
+    expect(response.body.message).to.equal('Article successfully deleted');
```

Comment:
This test is not enough to show an article was deleted from the article table. You need to query the database and ascertain the article is no longer in the article table.

Reply:
Thank you for this feedback.
The test in `Line 473` tests for success if an article is successfully deleted.
So when you try to delete the same article again, it should still return a success because the article is already deleted which was the intention of carrying out the delete operation in the first place.

To check that the article is no longer in the article table is tested on from `Lines 487 - 496`